SECURITY.md

```diff
@@ -37,8 +40,81 @@ We also support [Coordinated Vulnerability Disclosure (CVD)](https://en.wikipedi
 
-### Acknowledgements
 
-We appreciate and value the contributions of security researchers who take the time to responsibly disclose vulnerabilities. We can acknowledge your contribution (with your permission) once the fix is published, or you may remain anonymous if you prefer.
+### Acknowledgements and Bug Bounty Program 💰
+
+At this time, **Yeoman does not offer a bug bounty program** or monetary rewards for security vulnerability disclosures. However, we **greatly appreciate and recognize** the contributions of security researchers who responsibly disclose vulnerabilities.
+
+If you report a valid security issue, we will acknowledge your contribution (with your permission) in public advisories once the fix is published. Alternatively, you may remain anonymous if you prefer.
 
 Thank you for helping us maintain a secure environment across all [Yeoman](https://github.com/yeoman) repositories!
+
+## Threat Model
+
+The Yeoman threat model defines **trusted** and **untrusted** elements in the Yeoman ecosystem. All security vulnerabilities reported for any repository that the organization maintains are assessed within this context.
+
+### Elements Yeoman Does NOT Trust
+Yeoman **does not trust**:
+- **User Input**: CLI arguments, interactive prompts, and configuration files.
+- **Third-Party Generators**: Code from npm-installed generators unless officially maintained by Yeoman.
+- **Remote Templates & Assets**: Files fetched from external sources during scaffolding.
+- **Unvalidated Dependencies**: Packages installed dynamically during generator execution.
+
+Since Yeoman does not inherently trust these elements, generator authors should ensure proper validation, sanitization, and security checks in their own implementations. This includes handling user input safely, verifying dependencies, and ensuring templates are fetched from trusted sources.
+
+### Elements Yeoman Trusts
+Yeoman **trusts**:
+- **Node.js Runtime**: The JavaScript runtime and its security model.
+- **Operating System**: The OS environment where Yeoman runs.
+- **Official Yeoman Libraries**: Core packages such as `yo`, `generator-generator`, etc.
+- **npm Registry**: We rely on the npm registry for package distribution but do not inherently trust all packages. Each generator should be carefully evaluated before use. Users are encouraged to audit dependencies and be aware of potential supply-chain risks.
+- **Filesystem Integrity**: When reading/writing project templates, Yeoman assumes normal OS security constraints.
+
+## Examples of Security Vulnerabilities
+
+**Yeoman assesses vulnerabilities based on its threat model**. 
+
+If an issue arises due to insecure usage of untrusted components (e.g., an insecure third-party generator), it is generally the responsibility of the generator's maintainer. However, if Yeoman itself fails to properly mitigate risks when handling untrusted inputs, it may be considered a valid security issue.
```

Comment:
Nit: aligning the two clauses..

```suggestion
If an issue arises due to insecure usage of untrusted components (e.g., an insecure third-party generator), it is generally the responsibility of the untrusted component's maintainer. However, if Yeoman itself fails to properly mitigate risks when handling untrusted inputs, it may be considered a valid security issue.
```

Reply:
Thanks! :)